### PR TITLE
add sqldb module

### PR DESCRIPTION
**Pull request by sebastus**

### CODE REVIEW

**Comment by sebastus** [comment]
Bin - after you've had a chance to review and suggest changes, I'll squash.

File: lib/services/azuresqldb/client.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+var AZURE_RESOURCE_MANAGER_ENDPOINT = 'https://management.azure.com/';
+var MOONCAKE_RESOURCE_MANAGER_ENDPOINT = 'https://management.chinacloudapi.cn/';
+var AZURE_AD_ENDPOINT = 'https://login.windows.net/';
+var MOONCAKE_AD_ENDPOINT = 'https://login.chinacloudapi.cn/';
+
+var sqldbOperations = function (log, azure) {
+    this.log = log;
+    this.azure = azure;
+    this.resourceManagerEndpointUrl = (azure.environment == 'AzureChinaCloud') ? MOONCAKE_RESOURCE_MANAGER_ENDPOINT : AZURE_RESOURCE_MANAGER_ENDPOINT;
+    this.activeDirectoryEndpointUrl = (azure.environment == 'AzureChinaCloud') ? MOONCAKE_AD_ENDPOINT : AZURE_AD_ENDPOINT;
```
Comment (bingosummer) [inline lib/services/azuresqldb/client.js]: Is it tested in Mooncake?

File: test/testlist.txt [hunk 2]
```diff
@@ -23,5 +23,7 @@ unit/services/azureservicebus/deprovision-spec.js
 unit/services/azureservicebus/bind-spec.js
 unit/services/azureservicebus/unbind-spec.js
 
+unit/services/azuresqldb/provision-spec.js
```
Comment (bingosummer) [inline test/testlist.txt]: Please add more test cases for other operations.

File: test/testlist.txt [hunk 3]
```diff
@@ -23,5 +23,7 @@ unit/services/azureservicebus/deprovision-spec.js
...
+
 #integration/catalog.js
 #integration/lifecycle.js
```
Comment (bingosummer) [inline test/testlist.txt]: Could you please update the test matrix for lifecycle test?

A PR will be submitted by me to automate the lifecycle test. And I've submitted a code review to you. Please have a look.

File: lib/services/azuresqldb/client.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+
+    log.debug('sqldb client CTOR');
```
Comment (bingosummer) [inline lib/services/azuresqldb/client.js]: Could you please explain what `CTOR` is?

File: lib/services/azuresqldb/client.js (hunk 1 above)
Comment (sebastus) [inline lib/services/azuresqldb/client.js]: No - can I?

Comment (bingosummer) [inline lib/services/azuresqldb/client.js]: Sorry. I forgot that you didn't have a subscription in Mooncake. I'll test it.
However, I'm not sure whether SQL server is supported in Mooncake.

File: lib/services/azuresqldb/index.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    }
+  });
+  
+  // END
+  // these two sections are to minimize the amount of static data 
+  // that must be included in the developer's parameters file
+  //
+  
+  var cp = new cmdProvision(log, params);
+  log.debug('sqldb index: cmdProvision is newed up');
```
Comment (bingosummer) [inline lib/services/azuresqldb/index.js]: I think, `log.info` can be used to output the log.

File: lib/services/azuresqldb/index.js [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+      params.parameters.sqldbParameters.properties.edition = item.metadata.details.edition;
+      params.parameters.sqldbParameters.properties.requestedServiceObjectiveName = item.metadata.details.requestedServiceObjectiveName;
+    }
+  });
+  
+  // END
+  // these two sections are to minimize the amount of static data 
+  // that must be included in the developer's parameters file
+  //
+
```
Comment (bingosummer) [inline lib/services/azuresqldb/index.js]: A document is needed to explain how to set these parameters.

Comment (sebastus) [inline lib/services/azuresqldb/index.js]: I added it to my to do list.

**Comment by sebastus** [comment]
I need to get this PR merged as is - I've got some conflicts that I need to work out and it'll be much easier if I can do it after this PR.

File: lib/services/azuresqldb/index.js (hunk 5 above)
Comment (sebastus) [inline lib/services/azuresqldb/index.js]: I can change them.  Can hide the info messages using logule.json.

**Comment by sebastus** [comment]
I created another branch with my changes simplified into only a couple of commits.  I'll close this PR and open a new one.

**Pull request closed without merging**